contrib/externs/chrome_extensions.js

```diff
@@ -11214,3 +11214,96 @@ chrome.downloads.DeterminingFilenameEvent.prototype.hasListeners =
  * https://developer.chrome.com/extensions/downloads#event-onDeterminingFilename
  */
 chrome.downloads.onDeterminingFilename;
+
+
+/**
+ * @const
+ * @see https://developer.chrome.com/extensions/declarativeContent
+ */
+chrome.declarativeContent = {};
+
+
+/** @type {string} */
+chrome.declarativeContent.PageStateMatcherInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.ShowPageActionInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.SetIconInstanceType;
+
+
+/** @type {string} */
+chrome.declarativeContent.RequestContentScriptInstanceType;
+
+
+/**
+ * @typedef {{
+ *   pageUrl: (!UrlFilter|undefined),
```

Comment:
Internal reviewer pointed out that the docs do not say this field is a `UrlFilter`.
Instead, they explicitly list all of the fields.

Do you have any way of knowing whether the "real" requirement is for a `UrlFilter` or if the fields allowed may diverge in the future?

Reply:
@brad4d It is actually mentioned in the [documentation](https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher):

> (optional) pageUrl: Matches if the condition of the **UrlFilter** are fulfilled for the top-level URL of the page.

Other than that, I don't have access to the implementation of `PageStateMatcher` in order to check what is being used.

Reply:
@brad4d Here you have the answer 😃 
https://github.com/dart-gde/chrome.dart/blob/master/lib/gen/declarative_content.dart#L76

> `PageStateMatcher({UrlFilter pageUrl, List<String> css, bool isBookmarked})`

Reply:
Also, I found this documentation that seems to be generated from that code:
https://www.dartdocs.org/documentation/chrome/0.9.33/chrome.declarativeContent/PageStateMatcher-class.html

Reply:
I don't think those references count since they are for the dart wrapper around the API rather than the API itself.

Reply:
@brad4d I thought that project was from Google too.
If you check the [docs](https://developer.chrome.com/extensions/declarativeContent#type-PageStateMatcher) for `pageUrl` it says:
> Matches if the condition of the **UrlFilter** are fulfilled for the top-level URL of the page.

Reply:
I'm inclined to agree with you, so I've made the same point to the internal reviewer.

Reply:
Internal reviewer OK'd leaving this as-is.